File: WhaleModels/Whale.cs
using System;

namespace WhaleModels
{
    public class Whale
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Happiness { get; set; }
        public int Health { get; set; }
        public int Hunger { get; set; }
        public int PersonalityId { get; set; }
        Personality Personality { get; set; }
        public int SpeciesId { get; set; }
        Species Species { get; set; }
        public int Age { get; set; }
        public double Length { get; set; }
        public double Weight { get; set; }
        public string Sex { get; set; }
        public CalculateHappiness()
        {

        }
        public CalculateHealth()
        {

        }
        public CalculateHunger()
        {

        }

        public CalculateSizeAndWeight()
        {
            if (Age < Species.AdultAge)
            {
                //May need to modify
                Weight = Age * Species.SizeModifier + Weight;
                Length = Age * Species.SizeModifier + Length;
            }
        }

    }
}

File: WhaleAngular/src/app/models/whale.ts
import {Personality} from "./personality"
import {Species} from "./species"
export interface Whale {
    id: number,
    name: string,
    happiness: number,
    health: number,
    hunger: number,
    personalityID: number,
    personality: Personality,
    speciesId: number,
    species: Species,
    age: number,
    length: number,
    width: number,
    sex: string

}